src/cli/commands/snip.ts

```diff
@@ -19,15 +20,72 @@ interface SnipArgs extends MainArgs {
   destination: string;
   state?: string;
   ignore?: string | string[];
+  format?: string;
 }
 
-const handler = async ({
-  paths,
-  destination,
-  plugin,
-  state,
-  ignore,
-}: Arguments<SnipArgs>): Promise<void> => {
+export const doRst = async (
+  result: ParseResult
+): Promise<string | undefined> => {
+  const { source } = result;
+  if (
+    source.attributes["snippet"] === undefined ||
+    source.attributes["emphasize"] === undefined
+  ) {
+    return undefined;
+  }
+
+  // nasty hack to cover the suffixes/rst languages we use most often with Realm.
+  // TODO: switch to a better mapping
+  const rstLanguageMap: Map<string, string> = new Map([
+    [".js", "javascript"],
+    [".ts", "typescript"],
+    [".kt", "kotlin"],
+    [".java", "java"],
+    [".gradle", "groovy"],
+    [".m", "objectivec"],
+    [".swift", "swift"],
+    [".cs", "csharp"],
+    [".json", "json"],
+  ]);
+  const rstLanguage = rstLanguageMap.has(source.language)
+    ? rstLanguageMap.get(source.language)
+    : "text";
+
+  const rstHeader = ".. code-block::";
+  const rstEmphasizeModifier = ":emphasize-lines:";
+
+  const rstEmphasizeRanges: { start: number; end: number }[] = [];
+  for (const range of source.attributes["emphasize"]["ranges"]) {
```

Comment:
Really, we should have `export interface EmphasizeSourceAttributes` in EmphasizeCommand.ts and cast `source.attributes["emphasize"]` to that (or undefined) everywhere we want to use it. That allows you to use the attributes below with a bit more type safety and autocomplete support. That should be the pattern we follow (recognizing that we now use the word "attributes" in a few different contexts already...)

Reply:
Implemented this interface, and now using it (without the `| undefined`, by sheer personal preference).